helmfile/values/dex.yaml.gotmpl

```
@@ -75,6 +75,15 @@ config:
         - https://kibana.{{ .Values.global.baseDomain }}/auth/openid/login
         - https://kibana.{{ .Values.global.baseDomain }}/app/kibana
 {{ end }}
+{{ range $key, $value := .Values.dex.additionalStaticClients }}
+    - id: {{ $value.id }}
+      secret: {{ $value.secret }}
+      name: {{ $value.name }}
+      redirectURIs:
+      {{ range $key, $value := $value.redirectURIs }}
+        - {{ $value }}
+      {{ end }}
+{{ end }}
```

Wouldn't it work with just 
```suggestion
{{- toYaml .Values.dex.additionalStaticClients | nindent 4 }}
```

---

Just tested to change to this and it seemed to work fine. So I would like this solution instead, makes it easier if dex decides to change the structure of this data

---

Nice, that seems to work, but only when `additionalStaticClients != []`. Is there a smarter way for this rather than an `if len` as well?

---

> Nice, that seems to work, but only when additionalStaticClients != []. I'll put an if around it.

Ofcorse, nice catch!

> This does open things up to surprises a little bit when Dex updates and decides to change their structure in there. It might not be a problem 99% of the time, but still. In other words, Dex's abstraction is leaking into our own this way.

Sure, but that's kind of what we want. You should be able to read the documentation on dex and set it up the same way here. If we go this road and add our own layer of abstraction we might confuse people more if dex changes structure and we're not.

---

Alright, I can see that point of view as well :+1: the `toYaml` with an `if` around seems to work fine for both empty and non-empty arrays now.